src/sqlfluff/linter.py

```diff
@@ -881,11 +913,27 @@ def paths_from_path(
                     if fname.endswith(ext):
                         buffer.append(fpath)
 
+        if not ignore_files:
+            return sorted(buffer)
+
         # Check the buffer for ignore items and normalise the rest.
         filtered_buffer = []
+
         for fpath in buffer:
-            if fpath not in ignore_set:
+            if os.path.abspath(fpath) not in ignore_set:
                 filtered_buffer.append(os.path.normpath(fpath))
+            elif is_exact_file:
+                print(
+                    "WARNING: Exact file path %s was given but "
+                    "it was ignored by a %s pattern, "
+                    "re-run with `--not-ignore-files` to "
+                    "skip %s"
+                    % (
+                        path,
+                        ignore_file_name,
+                        ignore_file_name,
+                    )
+                )
```

Comment:
See comments about warnings. I think in #492 warnings should also end up being auto-formatted red so I'd love to use the logging framework where possible.

The reason for that is so that we can be more modular when people start importing `sqlfluff` as a module and not just running it through the cli.